### PR TITLE
feat(STADTPULS-312): HTTP Integration

### PR DESCRIPTION
**Pull request by ff6347**
This PR makes HTTP first class citizen in stadtpuls.com.
It cleans up the tests and removes global state in them.
It updates some settings on the docker setup and sql files.
It fixes some trailing slashes issues.
It adds some more info into the README which can be used later in the documentation.

It is not atomic. Sorry for that.

User can POST to their devices `/api/v2/devices/:deviceId/records` with their token.

**!Hint: The token is nothing you can use. ;)**

## With Curl 

```bash
curl --location --request POST 'http://localhost:4000/api/v2/devices/14/records' \
--header 'Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIwMDYxYmQ3OS1hMTQ5LTRjY2ItYjlmYS1lZDY0YmQ4YTcwMzAiLCJwcm9qZWN0SWQiOjYxLCJkZXNjcmlwdGlvbiI6Im15IGZhbmN5IHRva2VuIiwianRpIjoiMzY5ZjllNzYtMmRmMi00NzNkLTlkMGItNTgxNDg1N2RjNDk2IiwiaXNzIjoidGVjaG5vbG9naWVzdGlmdHVuZy1iZXJsaW4uZGUiLCJpYXQiOjE2MzEwMzEzMTJ9.2A4uLnNZRY0CixTUzCLB4CylBMj8VX6w0Yle0vklWW0' \
--header 'Content-Type: application/json' \
--data-raw '{
    "latitude": 52.483107,
    "longitude": 13.390679,
    "altitude": 30,
    "measurements": [
        1,
        2,
        3
    ]
}'
```

## With JS

```js
var myHeaders = new Headers();
myHeaders.append("Authorization", "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIwMDYxYmQ3OS1hMTQ5LTRjY2ItYjlmYS1lZDY0YmQ4YTcwMzAiLCJwcm9qZWN0SWQiOjYxLCJkZXNjcmlwdGlvbiI6Im15IGZhbmN5IHRva2VuIiwianRpIjoiMzY5ZjllNzYtMmRmMi00NzNkLTlkMGItNTgxNDg1N2RjNDk2IiwiaXNzIjoidGVjaG5vbG9naWVzdGlmdHVuZy1iZXJsaW4uZGUiLCJpYXQiOjE2MzEwMzEzMTJ9.2A4uLnNZRY0CixTUzCLB4CylBMj8VX6w0Yle0vklWW0");
myHeaders.append("Content-Type", "application/json");

var raw = JSON.stringify({
  "latitude": 52.483107,
  "longitude": 13.390679,
  "altitude": 30,
  "measurements": [
    1,
    2,
    3
  ]
});

var requestOptions = {
  method: 'POST',
  headers: myHeaders,
  body: raw,
  redirect: 'follow'
};

fetch("http://localhost:4000/api/v2/devices/14/records", requestOptions)
  .then(response => response.text())
  .then(result => console.log(result))
  .catch(error => console.log('error', error));
```

## With Python

```python
import requests
import json

url = "http://localhost:4000/api/v2/devices/14/records"

payload = json.dumps({
  "latitude": 52.483107,
  "longitude": 13.390679,
  "altitude": 30,
  "measurements": [
    1,
    2,
    3
  ]
})
headers = {
  'Authorization': 'Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIwMDYxYmQ3OS1hMTQ5LTRjY2ItYjlmYS1lZDY0YmQ4YTcwMzAiLCJwcm9qZWN0SWQiOjYxLCJkZXNjcmlwdGlvbiI6Im15IGZhbmN5IHRva2VuIiwianRpIjoiMzY5ZjllNzYtMmRmMi00NzNkLTlkMGItNTgxNDg1N2RjNDk2IiwiaXNzIjoidGVjaG5vbG9naWVzdGlmdHVuZy1iZXJsaW4uZGUiLCJpYXQiOjE2MzEwMzEzMTJ9.2A4uLnNZRY0CixTUzCLB4CylBMj8VX6w0Yle0vklWW0',
  'Content-Type': 'application/json'
}

response = requests.request("POST", url, headers=headers, data=payload)

print(response.text)
```

Closes STADTPULS-326 and STADTPULS-312

### CODE REVIEW

**Comment by ff6347** [comment]
Blocked by STADTPULS-326 externalId should be nullable.

**Comment by ff6347** [comment]
@julizet @dnsos @vogelino I know it is a lot, but I would love if you take a peak into the files and the description and ask me questions.

File: src/__test-utils/index.ts [hunk 1]
```diff
@@ -148,13 +174,25 @@ export const logout: (options: {
   }
 };
 
+export const betterDeleteUser: (userToken: string) => Promise<void> = async (
```
Comment (vogelino) [inline src/__test-utils/index.ts:177]: Better is always good 😝

**Review by vogelino — APPROVED** [review]
Great job @ff6347 ! I think everything is very legible and understandable. Maybe rename `betterDeleteUser` but aside from that it LGTM 👍

File: src/__test-utils/index.ts (hunk 1 above)
Comment (ff6347) [reply src/__test-utils/index.ts:177]: Okay. okay. Refactoed it.

**Comment by dnsos** [comment]
Really, really nice @ff6347! I POSTed a new record via HTTP now and it worked right away - amazing!

Also thanks for extending the docs!

**Pull request merged**